pkg/store/consul/consulutil/watch.go

```diff
@@ -317,6 +317,12 @@ func WatchDiff(
 				}
 				timer.Reset(2 * time.Second) // backoff
 				continue
+			} else if len(pairs) == 0 {
+				// Let's be safe and assume that we will never have an empty result.
+				// We will not be sending this to the watcher.
+				// Assume it means something happened with Consul, and try again.
+				timer.Reset(2 * time.Second) // backoff
+				continue
 			}
```

Comment:
while you're in here, maybe we should do a sanity check that `queryMeta.LastIndex` isn't less than `currentIndex`. Can save for a later PR if you desire